pkg/sql/backfill.go

```diff
@@ -1868,6 +1868,16 @@ func countIndexRowsAndMaybeCheckUniqueness(
 		if err != nil {
 			return 0, err
 		}
+		// Mark all the columns accessible if they are not. This is to fix a bug when attempting to create a unique,
+		// expression index on a REGIONAL BY table -- the accompanying expression column is inaccessible but later
+		// we check for uniqueness on this column via a SQL `SELECT` query, which of course then errors out. Such a fix
+		// is safe since this copy ('fakeDesc') is only used internally for validation.
+		// See https://github.com/cockroachdb/cockroach/issues/83076 for details.
+		for i, col := range fakeDesc.TableDesc().Columns {
+			if col.Inaccessible {
+				fakeDesc.TableDesc().Columns[i].Inaccessible = false
```

Comment:
Here `fakeDesc` is the return value of `tableDesc.MakeFirstMutationPublic` which is of type `catalog.TableDescriptor` which could be immutable as well as mutable. As it turns out, it's actually a mutable table descriptor right now but this is not guaranteed, and if it returns an immutable this code most likely won't work anymore.

I recommend getting a mutable descriptor beforehand:
```
mut, ok := fakeDesc.(*tabledesc.Mutable)
if !ok {
  mut = tabledesc.NewBuilder(fakeDesc.TableDesc()).BuildCreatedMutableTable()
}
for i := range mut.Columns {
  if col := &mut.Columns[i]; col.Inaccessible {
    col.Inaccessible = false
  }
}
desc = mut.ImmutableCopy()
```